### PR TITLE
CORE-610: Added a Java test for lambda Supplier interface

### PR DESCRIPTION
**Pull request by PaulHatcherR3**

### CODE REVIEW

File: quasar-kotlin/src/main/resources/META-INF/suspendable-supers [hunk 1]
```diff
@@ -0,0 +1,3 @@
+java.util.function.Supplier.*
```
Comment (chrisr3) [inline quasar-kotlin/src/main/resources/META-INF/suspendable-supers:1]: The Gradle build expects to _generate_ `META-INF/suspendable-supers` files using the `scanAndIInstrument()` function in `build.gradle`. Adding this `suspendable-supers` file of your own to `src/main/resources` will not "play nicely" with the generated file.

Comment (PaulHatcherR3) [reply quasar-kotlin/src/main/resources/META-INF/suspendable-supers:1]: Yes this is just a sandpit to try this out. Also as quasar-kotlin tests are loaded and instrumented by the JavaAgent.

The quasar documentation says that suspendable-supers can be added manually. So this is what we've tried in V5. This does seem to then work ok. So I'm not suggesting using the scanner as well. I'd rather just add these manually to suspendable-supers as needed.

Comment (chrisr3) [reply quasar-kotlin/src/main/resources/META-INF/suspendable-supers:1]: They _can_ be added manually, _except_ in the Quasar build which expects to generate them using an Ant task.

Comment (PaulHatcherR3) [reply quasar-kotlin/src/main/resources/META-INF/suspendable-supers:1]: Ok, that is understood, the fixes you have done are just for the quasar build. I've approved them.

**Pull request closed without merging**